api/src/main/java/org/pmiops/workbench/initialcredits/InitialCreditsBatchUpdateService.java

```diff
@@ -61,42 +57,59 @@ public InitialCreditsBatchUpdateService(
    * @param userIdList
    */
   public void checkInitialCreditsUsage(List<Long> userIdList) {
-    Set<String> googleProjects = workspaceDao.getWorkspaceGoogleProjectsForCreators(userIdList);
 
+    // Get the list of users from the database based on the userIdList.
+    List<DbUser> users = userDao.findUsersByUserIdIn(userIdList);
+
+    // This returns a map of google project id to its live cost in BigQuery.
+    Map<String, Double> terraWorkspacesLiveCosts = findTerraWorkspacesLiveCosts(userIdList);
+
+    // This is a map of user_id to its live total cost in BigQuery for VWB projects. We don't need
+    // the google project id here because we don't store it anywhere in the database and therefore
+    // it doesn't give us any information.
+    Map<Long, Double> vwbUserLiveCosts = findVwbUsersLiveCosts(users);
+
+    Set<DbUser> dbUserSet = Sets.newHashSet(users);
+
+    initialCreditsService.checkInitialCreditsUsageForUsers(
+        dbUserSet, terraWorkspacesLiveCosts, vwbUserLiveCosts);
+  }
+
+  private @NotNull Map<String, Double> findTerraWorkspacesLiveCosts(List<Long> userIdList) {
+    Set<String> googleProjects = workspaceDao.getWorkspaceGoogleProjectsForCreators(userIdList);
     Stopwatch stopwatch = stopwatchProvider.get().start();
     Map<String, Double> userWorkspaceCosts =
-        getAllWorkspaceCostsFromBQ().entrySet().stream()
+        initialCreditsBigQueryService.getAllTerraWorkspaceCostsFromBQ().entrySet().stream()
             .filter(entry -> googleProjects.contains(entry.getKey()))
             .collect(
-                Collectors.groupingBy(
-                    Entry::getKey, Collectors.summingDouble(Map.Entry::getValue)));
+                Collectors.groupingBy(Entry::getKey, Collectors.summingDouble(Entry::getValue)));
     Duration elapsed = stopwatch.stop().elapsed();
     log.info(
         String.format(
             "checkInitialCreditsUsage: Filtered %d workspace cost entries from BigQuery in %s",
             userWorkspaceCosts.size(), formatDurationPretty(elapsed)));
-
-    Set<DbUser> dbUserSet =
```

Comment:
I changed this to get all users by ID using `userDao.findUsersByUserIdIn(userIdList)`